### PR TITLE
feat(creator) 1.1.0: mainstream image/video providers, Anthropic/Gemini protocols, dialogue-gated video dispatch, expanded effects library, project copy & recreate, session status cards, 2GB uploads, and runtime reliability hardening

### PR DESCRIPTION
**Pull request by xuanrui-L**
## Description

This PR aggregates the latest Creator updates from the fork integration branch `dev/creator`. Highlights: Creator now reaches the mainstream model ecosystem — Anthropic/Gemini protocols for LLM/VLM and a broad set of image/video generation providers — dialogue becomes a first-class citizen of generated video prompts, the deterministic effects/motion library is significantly expanded, project copy & recreate lands, media limits are raised, plus a batch of field-incident-driven reliability fixes around runtime locks, durable interrupts, and scheduler stalls. Plugin version bumps to 1.1.0.

#### Mainstream image/video generation providers

- **Image**: Gemini (Nano Banana), Ark Seedream, BFL FLUX.2 and Ideogram providers with official reference/resolution contracts enforced before billing.
- **Video**: Veo 3.1 (Gemini API), Seedance 2.5 (Ark), MiniMax Hailuo, plus Kling and Vidu through both Bailian hosting and their official channels.
- Channel selection follows the saved protocol (user configuration), never the model name. Official request constraints are registered in the capability tables, rejected deterministically at submit time, and injected into specialist prompt guidance.
- The protocol dropdowns now list every media provider (presets registered with label keys and zh/en translations), so a saved protocol is never reset on load; the onboarding model guide and both Creator READMEs document the real provider matrix (including the Kling/Vidu Bailian-vs-official channels and official duration/reference limits), and a test pins the preset-reachability invariant.

#### Anthropic and Gemini protocols for LLM/VLM

- The plugin previously hardcoded OpenAI-compatible URL paths and body formats for all protocols; LLM/VLM clients are now protocol-aware (Anthropic `/v1/messages`, Gemini `generateContent`, OpenAI-compatible) with correct URLs, headers, body formats, and response parsing, including VLM content conversion (`image_url` → Anthropic image source, `inline_data` for Gemini).
- Connectivity probes branch per protocol; Gemini probes authenticate via the `key=` query parameter, Anthropic sends `x-api-key` only when configured, and keyless OpenAI-compatible providers are supported (Authorization headers only when a key exists, dynamically provided protocols survive reloads). Gemini VLM downloads remote media and inlines it as base64, since `file_data.file_uri` only accepts Files API or `gs://` resources. Protocol classification is centralized so probes, models, and the agent runtime agree.
- Token Plan support: a dynamic provider dropdown merges host-configured and free-tier providers (free providers labeled accordingly, custom entry always available); image generation returns `source_url` to avoid re-upload, and video references use HTTP(S) URL passthrough instead of download+upload.
- Model-config errors now name the protocol, model, and endpoint (credentials redacted), the upstream HTTP status with a response excerpt, and a status-specific remediation hint, instead of a bare "model call failed".

#### Session status on project listing cards

Project cards in the listing now surface live session status, so running/waiting/failed sessions are visible without opening the project.

#### Dialogue reaches the video provider

- **Prompt-compilation doctrine.** Field runs produced silent films despite written scripts: per-shot dialogue was planned, then a mood summary was committed as `video_prompt` and dispatched verbatim. The coordinator system prompt now treats dialogue as the default for narrative scenes (silence requires an explicit directorial note) and requires `video_prompt` to be compiled shot-by-shot, quoting each spoken line with performance notes and a sound-design line.
- **Dialogue-coverage gate in the work graph.** The video node stays GATED until `video_prompt` quotes every planned spoken line (speaker prefixes and stage directions excluded); an empty `video_prompt` is now a named model-required gap instead of a dispatch-time `ValidationError`.
- **Per-element density gate.** `R2VCreation.min_dialogue_ratio` (default 0.3 ≈ 1 line per 2–3 shots) gates short-drama elements whose shots lack dialogue; exemptions via `ratio=0` opt-out or an explicit "有意静默" note in the narrative. Regression tests cover all branches.
- **Three scheduler stall fixes from its dogfood.** (1) The work graph mirrors the project-wide lineup gate and READY nodes whose dispatch died before task admission reopen within a bounded transient budget instead of being poisoned by the dispatch ledger forever. (2) Provider "weather" outliving the retry budget gets one further retry per 5-minute cooldown window, hard-capped at 6, so pipelines self-heal while paid spend stays bounded. (3) Startup recovery of zombie RUNNING tasks now fail-closes the record it actually iterated.

#### Expanded effects library with content-type matching and keyword overlay

Large expansion of deterministic motion blueprints and motion design, plus content-type-aware effect matching and narrowed keyword-overlay detection in the editing pipeline and AI-editing-director prompt.

#### Project copy and recreate

New project-copy/recreate API route plus HomePage/HeroComposerCard flows and a dedicated recreate store, so users can fork an existing project as the starting point of a new one.

#### 2GB local video uploads

Raised the local video upload limit from 100MB to 2GB to match DashScope VLM's accepted input size, with the backend error message in English for i18n parity.

### Runtime reliability: locks, interrupts, and diagnostics

- **Scene-lock deadlock and compose-gate fixes.** Resolved the scene-lock deadlock in the pre-compose review path, fixed an opacity-repair cache-key bug and stale-task handling (blocking VLM calls) in pre-compose auto-review, and ensured the `paw_trail` motion motif has visible content in the first frame.
- **Unwedge durable interrupts and cut lock starvation.** From a field incident where a failed mainline run parked its Session in `INTERRUPT_REQUESTED` forever while r2v generations died on 10s lock timeouts: `session_store` now reads the durable event-tail seq in O(1) instead of replaying/revalidating the whole events stream under the exclusive project lock; terminal cleanup retries Runtime-lock timeouts with backoff so a failure cascade can no longer strand a run without a terminal status; a durable interrupt pointing at an ownerless RUNNING run past a 120s stall window now reclaims it (conditional transition to FAILED).
- **Per-model image/video reference budget.** The reference budget is injected per configured model — qwen-image only accepts 0–3 image content items per call (observed live: HTTP 400 with 4 references); other providers keep the historical budget of 5. Video reference budgets are enforced as well, alongside hardened runtime diagnostics.
- **Admission-gate locking.** The project lifecycle/runtime locks gained an admission gate that stops the UI polling reader stream from starving writers into `LockTimeoutError`; frontend polling is additionally slowed and paused on hidden tabs (`visiblePolling`).

### Compose/render correctness

- Fixed subtitle overflow in overlay rendering, fallback box sizing, the edge-contact threshold for overlay placement, and the color-grade filter application; the segment cache is keyed by `rendererVersion` v7 so stale segments from earlier renderers are invalidated.
- Caption font size is normalized across overlays with different box dimensions, so captions read consistently regardless of the overlay box they live in.

### Secure-context and session UX

- **Pure-JS SHA-256 fallback** keeps frontend CAS edits working on `http`+IP deploys where `crypto.subtle` is unavailable (fixes the timeline "digest" crash).
- **Backward message paging.** Session history pages backward (`tail`/`before` + `nextBefore`); AgentDock loads the newest page first and auto-loads older history on scroll-up with viewport anchoring.
- **Authoritative `[Image N]` reference-order preview** endpoint with workbench badges; the materials picker groups uploads vs artifacts and filters videos for HappyHorse r2v; committed numbering is hidden while the local draft is dirty, and reference rows resolve kind-correct refs (`asset-version:` for uploads, `artifact-version:` for artifacts).

### Test suite maintenance

Backend pytest and frontend vitest suites were slimmed by ~45% while keeping regression coverage: fragmented small files merged, duplicate weak tests deleted (one strongest path per behavior kept, plus every incident regression / fail-closed / concurrency guard), parametrize tables trimmed to boundary cases, and giant fixtures shrunk.

**Security Considerations:** The image/video reference budgets and provider request contracts are fail-closed (over-budget or over-limit requests are rejected before paying for a model call). Error diagnostics redact credentials (API keys shown only as `<set>`/`<empty>`, URLs redacted). The SHA-256 fallback is pure JS with no network dependency.

## Type of Change

- [x] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Documentation
- [ ] Refactoring

## Component(s) Affected

- [x] Core / Backend (app, agents, config, providers, utils, local_models)
- [ ] Console (frontend web UI)
- [ ] Channels (DingTalk, Lark, QQ, Discord, iMessage, etc.)
- [ ] Skills
- [ ] CLI
- [ ] Documentation (website)
- [x] Tests
- [ ] CI/CD
- [ ] Scripts / Deploy

## Checklist

- [x] I ran `pre-commit run --all-files` locally and it passes
- [x] If pre-commit auto-fixed files, I committed those changes and reran checks
- [x] I ran tests locally (`pytest` or as relevant) and they pass
- [x] Documentation updated (if needed)
- [x] Ready for review

## Testing

- Creator backend suite: `python -m pytest -q` from `plugins/apps/qwenpaw-creator/backend` — covers the protocol-aware LLM/VLM/image/video clients and probes, provider capability contracts, dialogue gates, scheduler stall fixes, lock/interrupt recovery, reference budgets, project copy/recreate API, and the expanded motion blueprints.
- Plugin packaging: `pytest tests/unit/plugins/test_generate_plugin_metadata.py`.
- Creator UI: `npx vitest run` from `plugins/apps/qwenpaw-creator/ui` (incl. the preset-reachability invariant, sha256 fallback and R2V workbench `[Image N]` tests) plus `tsc --noEmit`.
- Live dogfood: the dialogue gate and scheduler fixes were caught and validated on real dogfood projects (silent-film reproduction, 96k-event lock-starvation incident, INTERRUPT_REQUESTED wedge).

## Evidence

```bash
$ cd plugins/apps/qwenpaw-creator/backend && python -m pytest -q
911 passed, 2 skipped, 21 deselected in 38.67s
# (the 2 skips are optional heavy deps: geopandas/trimesh)

$ cd <repo-root> && python -m pytest tests/unit/plugins/test_generate_plugin_metadata.py -q
..........................                                               [100%]
26 passed in 2.13s

$ cd plugins/apps/qwenpaw-creator/ui && npx vitest run
 Test Files  46 passed (46)
      Tests  246 passed (246)

$ cd plugins/apps/qwenpaw-creator/ui && npx tsc --noEmit
(exit 0)

$ pre-commit run --from-ref upstream/main --to-ref HEAD
check python ast.........................................................Passed
check docstring is first.................................................Passed
check json...............................................................Passed
fix python encoding pragma...............................................Passed
detect private key.......................................................Passed
trim trailing whitespace.................................................Passed
Add trailing commas......................................................Passed
black....................................................................Passed
flake8...................................................................Passed
pylint...................................................................Passed
prettier.................................................................Passed
```

## Additional Notes

- The net diff of this PR against `main` is 320 files, +28004 −47379, all under `plugins/apps/qwenpaw-creator/` (the large deletion count comes from the test-suite slimming pass). Host files outside the plugin are untouched.
- The Creator plugin version is bumped to 1.1.0 (`plugin.json` + UI `package.json`/`package-lock.json`).
- The branch includes a merge of the latest `upstream/main`; conflicts with previously squashed Creator content and the datapaw release exclusion were resolved by keeping the newer dev/creator behavior (dialogue gate in the system prompt, `video_missing` surfacing in the work graph) while adopting upstream's datapaw release exclusion and `pack_requires` docs.

### CODE REVIEW

File: plugins/apps/qwenpaw-creator/backend/utils/logger.py [hunk 1]
```diff
@@ -74,20 +76,39 @@
...
+            for handler in self._targets.values():
+                handler.close()
+            self._targets.clear()
         super().close()
 
+    def close_project(self, project_id: str) -> None:
+        """Close and forget one Project target before its tree is deleted."""
+
+        try:
+            project_root = (self.data_root / project_id).resolve(strict=False)
```
[the remark this answers is not in the document]
Comment (xuanrui-L) [reply plugins/apps/qwenpaw-creator/backend/utils/logger.py]: Fixed in 1855a7b3: close_project now strips directory components with os.path.basename (and rejects . / ..) before composing the log path, keeping the existing containment check, so a crafted project id can never point the handler lookup outside the data root.

File: plugins/apps/qwenpaw-creator/backend/api/dependencies.py [hunk 2]
```diff
@@ -99,31 +133,227 @@
...
         content={
             "code": error.code,
             "message": error.message,
             "retryable": error.retryable,
             "details": error.details,
+            "errorId": report["errorId"],
+            "traceId": trace_id,
+            "requestId": request_id,
+            "occurredAt": datetime.now(UTC).isoformat(),
+        },
```
[the remark this answers is not in the document]
Comment (xuanrui-L) [reply plugins/apps/qwenpaw-creator/backend/api/dependencies.py:201]: Addressed in 1855a7b3: the exposed branch only admits AppError-style domain errors whose messages are curated at raise time (credentials redacted via redact_url, no stack frames); truly unexpected exceptions take the INTERNAL_ERROR branch which never exposes exception text. Documented with an inline codeql[py/stack-trace-exposure] suppression rationale at the mapping site.

File: plugins/apps/qwenpaw-creator/backend/services/file_agent_runtime/driver.py [hunk 3]
```diff
@@ -5753,23 +5977,68 @@
...
+                            },
+                        )
+                    except AgentRunStateConflict:
+                        # A live owner moved the run after all; it now
+                        # serves the interrupt.
+                        return
+                    logger.warning(
+                        "reclaimed orphaned RUNNING run for durable "
+                        "interrupt: project=%s run=%s",
+                        project_id,
```
[the remark this answers is not in the document]
Comment (xuanrui-L) [reply plugins/apps/qwenpaw-creator/backend/services/file_agent_runtime/driver.py]: Fixed in 1855a7b3: user-provided values (project_id, run_id) are now passed through _log_safe, which neutralizes CR/LF before logging — same pattern already used across the plugin.

File: plugins/apps/qwenpaw-creator/backend/services/file_agent_runtime/driver.py [hunk 4]
```diff
@@ -5783,7 +6052,16 @@
                 actor="user",
                 payload={"reason": reason},
             )
+            self._interrupt_stalls.pop(project_id, None)
         except Exception:
+            # The next reconcile pass retries; keep the failure visible —
+            # this path being silent hid a permanently wedged Session.
+            logger.warning(
+                "idle interrupt cleanup failed: project=%s reason=%s",
+                project_id,
```
[the remark this answers is not in the document]
Comment (xuanrui-L) [reply plugins/apps/qwenpaw-creator/backend/services/file_agent_runtime/driver.py]: Fixed in 1855a7b3: project_id and the user-supplied interrupt reason are wrapped with _log_safe (CR/LF neutralization) before logging.

**Review by cuiyuebing — APPROVED** [review]
LGTM
